src/khoj/routers/api_chat.py

```diff
@@ -532,6 +591,27 @@ async def send_rate_limit_message(message: str):
             await conversation_command_rate_limiter.update_and_check_if_valid(websocket, cmd)
             q = q.replace(f"/{cmd.value}", "").strip()
 
+        if conversation_commands == [ConversationCommand.Summarize]:
+            file_filters = conversation.file_filters
+            if len(file_filters) == 0:
+                await send_complete_llm_response(
+                    "No files selected for summarization. Please add files using the section on the left."
+                )
+            elif len(file_filters) > 1:
+                await send_complete_llm_response("Only one file can be selected for summarization.")
+            else:
+                try:
+                    file_object = await FileObjectAdapters.async_get_file_objects_by_name(file_filters[0])
+                    contextual_data = " ".join([file.raw_text for file in file_object])
+                    if not q:
+                        q = "Create a general summary of the file"
+                    await send_status_update(f"**🧑🏾‍💻 Constructing Summary Using:** {file_object[0].file_name}")
+                    response = await extract_relevant_summary(q, contextual_data)
+                    await send_complete_llm_response(str(response))
+                except Exception as e:
+                    await send_complete_llm_response(f"Error summarizing file.")
```

Comment:
Add a log line here to log the exception, like
```suggestion
            logger.error(f"Error summarizing file for {user.email}: {e}")

```